django-stubs/test/testcases.pyi

```diff
@@ -216,7 +218,11 @@ class TransactionTestCase(SimpleTestCase):
 class TestCase(TransactionTestCase):
     @classmethod
     def setUpTestData(cls) -> None: ...
-    def captureOnCommitCallbacks(cls, *, using: Optional[str] = ..., execute: bool = ...): ...
+    @classmethod
+    @contextmanager
+    def captureOnCommitCallbacks(
+        cls, *, using: Optional[str] = ..., execute: bool = ...
+    ) -> Generator[Sequence[Callable[[], Any]], None, None]: ...
```

Comment:
We should be specific and use `List` as the return type. This way tests that use list-specific operations still type check fine.

```suggestion
    ) -> Generator[List[Callable[[], Any]], None, None]: ...
```

Reply:
Yeah, it would be nice to lock down weird operations with type hints, but to avoid type hints introducing unnecessary churn to projects, I think it's most pragmatic to document Django as-is.

Reply:
Yes, +1

This is the same we do in typeshed.